### ISSUE TITLE
type B = | A does not produce an error

### ISSUE THREAD

**Issue by jakewilson** [microsoft/TypeScript#36415]
Putting a pipe operator `|` at the beginning of a type expression like `type B = | A` does not produce an error. It does produce an error at the beginning of an assignment, like `const a = | 5;`.

**TypeScript Version:**  3.7.x-dev.201xxxxx

**Search Terms:**
pipe operator expression beginning

**Code**

```ts
interface A {
    id: number;
}

type B = | A;

const b: B = { id: 4 };
```

It also works with multiple types:
```ts
interface A {
    id: number;
}

interface C {
    name: string;
}

type B = | A | C;

const b: B = { name: 'jake' };
```
**Expected behavior:**
Should throw an error because the first pipe is unexpected.

**Actual behavior:**
Compiles - B is of type A in the first example, B is of type A | C in the second.

**Playground Link:** 
http://www.typescriptlang.org/play/index.html?ssl=7&ssc=1&pln=7&pc=24#code/JYOwLgpgTgZghgYwgAgILIN4Chm+cAEwC5kQBXAWwCNoBuLAXyyzAE8AHFAIWQF5kAPmnpYEAexABnMMioke-DPmLIALMga0gA

**Related Issues:**

**Comment by orta** [comment]
This is on purpose to support writing a union in a [format like this](http://www.typescriptlang.org/play/index.html#code/C4TwDgpgBAglC8UDeBfAUKSUBCDns2gGE9U0NxoBZASwA8E0pmoAfWJl97T59o8mgDGAewB2AZ2BQARgC4otBoiRQaAEwUAWKCgDcaIA):

```ts
type A = {}
type B = {}
type C = {}

type Mix =
    | A
    | B
    | C

const b: Mix = { id: 4 };
```

**Comment by jakewilson** [comment]
oh cool, so it's just to look nicer?

**Comment by orta** [comment]
Yeah, it's a bit like what folks do with trailing commas. [Prettier uses that style for example](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAsgSwA9sBeAH2wEEAZaAcwBVM4AdKbD7CgIVqkeZtOXbAGE+ArEM4VswbAEMk2AM4wATvn4AabGGVQArgFsARnHW64ytZv7YAvtLZtIUNdlPKCxEnOz4ACbKACyOANwg2iAQGDD40CrIoArq6hAA7gAKqQhJKAoAbhBBUSCm6gpgANZwMADKGFVadMgahnDRABYwxgA2AOpd+PAqTWBw9Xkj+IUjaMjgKknRWioWMFmVdMYKyABmCn3r0QBWKoTclTV19QrGcNRacAdHJyDnhPUtfXAAioYIPBXsdOiAmup1upFqYFOY+mUMHYYAMgjAusgABwABmiSIg6wGlQwiyRcChhRe0QAjoD4FtYvkQAoVABaKBwOCBLlldRwWn4PlbBQ7PZIQ6g6LrYz4NrqDpSn7-OkvcVvMEwOGowLo5AAJmiGgU+D6LVEEGMu0WUGgVJAhnWDDh+Ql70KHQAklBubB6mBNHFKN76uhfiD1g4HEA).
